andela/travel_tool_back
# #161758633 disable change room if user is checked out

#### What does this PR do?
-Disable travel admin min from changing requester's room once they have checked out
#### Description of Task to be completed?

-Add validator that checks if a user is checked out
-disable travel admin from re-assigning room if user is checked out
-add test case for this validation

#### How should this be manually tested?

- git clone this repo - `git clone https://github.com/andela/travel_tool_back.git`

- Enter the project folder - `cd travel_tool_back`

- git checkout this branch - `bg-disable-change-room-161758633`

- run migrations using `yarn: dbrollmigrate`

- update dependencies using `yarn start`

- Start the application using `make start` on the backend and frontend

- Navigate to travela application on the browser and login. Once you have successfully logged in retrieve your jwt token from the cookies tab

- While still on the application create a new request and have it approved. Navigate the the `Residence tab`, click on `Check in` and subsequently `Check out`

- Paste your jwt token on the headers of a postman request with a `PUT` method to the url `http://localhost:5000/api/v1/trips/:trip id/: room id`

#### Any background context you want to provide?
You need to assign yourself the roles of `manager` and `Travel Admin` on the backend to be able to test this.
#### What are the relevant pivotal tracker stories?
[161758633](https://www.pivotaltracker.com/story/show/161758633)
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/37342753/48422770-67492380-e770-11e8-8220-f5962291ee53.png)

#### Questions:
N/A
